mlir/lib/Transforms/ViewOpGraph.cpp

```diff
@@ -261,9 +273,18 @@ class PrintOpPass : public impl::ViewOpGraphBase<PrintOpPass> {
         if (printControlFlowEdges && prevNode)
           emitEdgeStmt(*prevNode, nextNode, /*label=*/"",
                        kLineStyleControlFlow);
+
+        // If we are at first iteration, save the first operand node for the
+        // incoming edges.
+        if (not prevNode) {
+          blockFirstNodeMap[&block] = nextNode;
+        }
         prevNode = nextNode;
       }
-    });
+
+      // Save the last operation for the outgoing edges.
+      blockLastNodeMap[&block] = *prevNode;
+    }, nodeName);
```

Comment:
Same as above.

Reply:
This suffers from the same problem of the situation explained above.